### PR TITLE
Change documented x509 salt mine example, fixes #56142

### LINKED ISSUE #56142: x509 Documentation -> mine.send not working
### Description of Issue
I'm trying to set up a CA based on the example provided in https://docs.saltstack.com/en/latest/ref/states/all/salt.states.x509.html and also using https://backbeat.tech/blog/using-saltstack-for-internal-ssl-certificates/ as a reference.

AFAICT the documented mine.send for the ca.crt is not working. I find two issues: 
* The onchange does not trigger if the ca.crt already exists
* After removing the onchange, I get a `The following arguments are missing: m_name`

### Setup
The setup is basically the example provided in the documentation with no changes.

### Steps to Reproduce Issue
Follow the documentation.

### Versions Report
```
Salt Version:
           Salt: 3000
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.7.3
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.7.3 (default, Dec 20 2019, 18:57:59)
   python-gnupg: Not Installed
         PyYAML: 3.13
          PyZMQ: 17.1.2
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist: debian 10.3 
         locale: UTF-8
        machine: x86_64
        release: 4.19.0-8-amd64
         system: Linux
        version: debian 10.3 
```

### PR DESCRIPTION
**Pull request by svenseeberg**
### What does this PR do?
* The used module.run syntax is outdated. Replace with  minion.d configuration that now includes mine_functions.
* Rename the minion.d/signing_policies.conf to x509.conf  to represent the broader scope of the file.
* Additional fix: the allowed peer executable functions must not contain the module name.

### What issues does this PR fix or reference?
This pull request fixes #56142 

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by cmcmarrow — APPROVED** [review]

**Comment by cmcmarrow** [comment]
Thanks @svenauhagen

**Review by DmitryKuzmenko — APPROVED** [review]

**Comment by DmitryKuzmenko** [comment]
@svenseeberg could you please resolve merge conflicts?

File: salt/states/x509.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
     peer:
       .*:
-        - x509.sign_remote_certificate
+        - sign_remote_certificate
```
Comment (joni1993) [inline salt/states/x509.py:27]: I tried the configuration this way but it refuses with `salt.exceptions.SaltInvocationError: ca_server did not respond salt master must permit peers to call the sign_remote_certificate function.`
When using `x509.sign_remote_certificate` everything works fine

Comment (svenseeberg) [reply salt/states/x509.py:27]: That's interesting. I had in the end both versions, with and without the `x509`, in the config. And I can confirm that now the x509 is required. I'm not sure if something changed or I made a mistake here. But I tested this pretty thoroughly back then.

So I guess we should revert this commit or update the documentation again.

Comment (svenseeberg) [reply salt/states/x509.py:27]: Created the PR: https://github.com/saltstack/salt/pull/57455

**Pull request merged**